design/aws-ecs/aws-ecs-tagging-support.md

```diff
@@ -0,0 +1,196 @@
+# AWS ECS - Tagging Support and Tagging API Change
+
+Amazon ECS supports tagging for all the ECS resources (see [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-using-tags.html)). Note that ECS provides a property of ECS service named `PropagateTags` to support propagating tags from either task definitions or services to tasks (tasks cannot be tagged using CDK without setting this parameter). Currently this property is not set in CDK so that the tags are not propagated by default.
+
+In addition, `EnableECSManagedTags` is provided as another property so that ECS managed tags can be supported and used to tag ECS tasks. However, currently users cannot set `EnableECSManagedTags` and the default value is `false`.
+
+Additionally, CDK support tagging and can cascade tags to all its tagable children (see [here](https://docs.aws.amazon.com/cdk/latest/guide/tagging.html)). The current CDK tagging API is shown below:
+
+``` ts
+myConstruct.node.applyAspect(new Tag('key', 'value'));
+
+myConstruct.node.applyAspect(new RemoveTag('key', 'value'));
+```
+
+As we can see, the current tagging API is not nice and intuitive enough for users which needs to be improved.
```

Comment:
This comment feels un-substantiated to me. What is not nice and intuitive about this API?

Reply:
Does this look good to you?